sql/core.go

```diff
@@ -370,14 +371,56 @@ type SystemVariableRegistry interface {
 	NewSessionMap() map[string]SystemVarValue
 	// GetGlobal returns the global value of the system variable with the given name
 	GetGlobal(name string) (SystemVariable, interface{}, bool)
+	// GetResetVal returns the default value of the system variable with the given name.
+	// This is not the current GLOBAL value. It's used for postgres config parameters only.
+	GetResetVal(name string) (SystemVariable, interface{}, bool)
 	// SetGlobal sets the global value of the system variable with the given name
 	SetGlobal(name string, val interface{}) error
 	// GetAllGlobalVariables returns a copy of all global variable values.
 	GetAllGlobalVariables() map[string]interface{}
 }
 
-// SystemVariable represents a system variable.
-type SystemVariable struct {
+// SystemVariable is used to handle mysql system variables and postgres configuration variables interchangeably.
+// TODO: mysql sv and postgres cp should not be mixed.
+type SystemVariable interface {
+	// GetName returns the name of the sv. Case-sensitive.
+	GetName() string
+	// SetName sets the name of the sv.
+	// TODO: it should not happen, but we set the name to lower-case string?
+	SetName(string)
+	// GetType returns the type of the sv.
+	GetType() Type
+	// SetDefault sets the default value of the sv.
+	SetDefault(any)
+	// GetDefault returns the default value of the sv.
+	GetDefault() any
+	// GetValue returns the value
+	GetValue(a any) (any, bool)
```

Comment:
It's not clear from the documentation or parameter name what `a` should be, or why it's necessary. Looking at `MysqlSystemVariable`, it looks like it's a default value for if no actual value is set? It would be much clearer to have `GetValue() (any, bool)`, and another `GetValueWithDefault(default any) (any, bool)`, if that's what you're trying to do here. If not, then it still needs to be clearer.

Reply:
This was bit confusing to me as well to differentiate `GetValue` from `GetDefault`. I forgot to complete the docs here.
What I understood is that current GLOBAL is the default value in MySQL. So, setting the session variable to default is the same as setting it to current GLOBAL value. Then, setting the global variable to default is basically no-op.
However, in Postgres, `SET ... TO DEFAULT` = `RESET`, setting parameter to default means the same as `RESET`-ing the parameter to its original value.
The confusion is that `GetGlobal()` returns both the current global value and the default value in initial GMS version, since they are the same thing. But what we need for Postgres is differentiate the cases of `GetGlobal`, current SESSION value, and `GetDefault`, the original default value.